### PR TITLE
Strain input psd

### PR DESCRIPTION
**Pull request by micamu**
Add an option to strain.py to generate a fake strain from a given ASD file.

### CODE REVIEW

File: pycbc/strain.py [hunk 1]
```diff
@@ -328,6 +328,20 @@ def from_cli(opt, dyn_range_fac=1, precision='single'):
             logging.info("Converting to float32")
             strain = (dyn_range_fac * strain).astype(pycbc.types.float32)
 
+    if opt.fake_strain_from_file:
+        logging.info("Generating Fake Strain from ASD file")
```
Comment (ahnitz) [inline pycbc/strain.py]: Can you avoid the code duplication of the next few lines? Perhaps always calculate these quanitties before the if statements.

**Comment by ahnitz** [comment]
Besides for the one code comment, this looks good. Have you had a chance to test it?

**Comment by spxiwh** [comment]
Miriam, as it is this patch would not allow one to use injections if using this new option. Why not wrap this into the fake-strain option to avoid duplication and avoid this?

**Comment by ahnitz** [comment]
Ian is right. I still think we want a separate option for clarity. Miriam move the injection part to after the fake strain from file, and put it in an if statement that is true if either fake-strain for fake-strain-from-file is given.

File: pycbc/strain.py [hunk 2]
```diff
@@ -562,6 +569,10 @@ def insert_strain_option_group_multi_ifo(parser):
                             action=MultiDetOptionAction, metavar='IFO:SEED',
                             help="Seed value for the generation of fake "
                             "colored gaussian noise")
+    data_reading_group.add_argument("--fake-strain-from-file", nargs="+",
+                            action=MultiDetOptionAction, metavar='IFO:FILE',
+                            help="File containing ASD for generating fake "
+                            "noise from it.")
```
Comment (spxiwh) [inline pycbc/strain.py]: You get the:

seal_of_approval

for updating the multi-ifo option group as well!

Comment (micamu) [inline pycbc/strain.py]: Just following Alex's instructions :)

**Comment by micamu** [comment]
Tested with pycbc_inference, it worked both with the options --asd-file and --fake-strain-from-file, and with the options --psd-model and --fake-strain

**Pull request merged**